core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java

```diff
@@ -26,9 +26,24 @@ public class LogErrorHandler implements Handler<RoutingContext> {
   public void handle(final RoutingContext failureContext) {
 
     if (failureContext.failed()) {
-      LOG.error(
-          String.format("Failed request: %s", failureContext.request().absoluteURI()),
-          failureContext.failure());
+      String requestUri;
+      try {
+        requestUri = failureContext.request().absoluteURI();
+      } catch (final NullPointerException e) {
+        // absoluteURI can throw an NPE if host or port is missing
+        requestUri = "Error in calculating request URI";
```

Comment:
More descriptive error would be nicer here along line of invalid URI due to missing host or port.

Reply:
sure. This NPE happens when invalid host header is involved from the client. Latest vertx has started throwing an NPE. I'll update the error description.

Reply:
refactored it a bit.